### PR TITLE
[SYCLomatic] Avoid collision of numeric limit max with windows.h

### PR DESCRIPTION
**Pull request by LU-JOHN**
Signed-off-by: Lu, John [john.lu@intel.com](mailto:john.lu@intel.com)

### CODE REVIEW

File: clang/runtime/dpct-rt/include/kernel.hpp.inc [hunk 1]
```diff
@@ -128,7 +128,7 @@ static inline fs::path write_data_to_file(char const *const data, size_t size) {
   std::error_code ec;
 
   if (sizeof(size_t) >= sizeof(std::streamsize) &&
-      size > std::numeric_limits<std::streamsize>::max())
+      size > (std::numeric_limits<std::streamsize>::max)())
```
Comment (0x12CC) [inline clang/runtime/dpct-rt/include/kernel.hpp.inc:131]: Why is this change needed?

Comment (jzc) [reply clang/runtime/dpct-rt/include/kernel.hpp.inc:131]: `windows.h` defines the macros min/max, so parentheses are needed to prevent macro invocation

**Review by jzc — APPROVED** [review]

**Review by 0x12CC — APPROVED** [review]
